### PR TITLE
Add CSRF mitigations to the forms

### PR DESCRIPTION
**Pull request by JoelClemence**
* Apply CsrfProtectionMiddleware to the application and add CSRF token to each form

### CODE REVIEW

File: src/views/router-views/email/update-submitted.njk [hunk 1]
```diff
@@ -34,6 +34,11 @@
 
     {% if true === returnToConfirmationStatement %}
         <form action="" method="post">
+            {{
```
Comment (ch-ncolley) [inline src/views/router-views/email/update-submitted.njk:37]: Non-blocking (just something to consider in the future): Nunjucks style here a bit more consistent with the other Macros having it like"

```njk
{{ csrfTokenInput({ csrfToken: csrfToken }) }}
```

or 


```njk
{{ csrfTokenInput({
    csrfToken: csrfToken 
}) }}
```

**Review by ch-ncolley — APPROVED** [review]
I've not worked on this application before, so my review is maybe not so strong, but I have had a look over and it looks like sensible changes and nothing obviously amiss.

**Comment by ch-ncolley** [comment]
@JoelClemence I meant non-blocking as in my feedback on that syntax was not a blocking comment, that you could ignore it has not functional difference - sorry

**Comment by JoelClemence** [comment]
> @JoelClemence I meant non-blocking as in my feedback on that syntax was not a blocking comment, that you could ignore it has not functional difference - sorry

I get you. No worries, thanks.

File: test/src/middleware/csrf_error_middleware.unit.ts [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+      {},
+      nextMock
+    );
+
+    expect(nextMock).toHaveBeenCalledWith(notCsrfError);
+  });
+
+  it("renders 403 error when CSRF error", () => {
+    const nextMock = jest.fn();
+    const notCsrfError = new CsrfError("Token mismatch");
```
Comment (NeilShadrach) [inline test/src/middleware/csrf_error_middleware.unit.ts:27]: Isn't this misnamed? csrfError?

**Pull request merged**